test/jdk/javax/rmi/ssl/SSLSocketParametersTest.java

```diff
@@ -116,174 +130,122 @@ public ServerSocket createServerSocket(int port) throws IOException {
+
+            /* client side dummy_ciphersuite */
+            case 6 -> {
                 System.setProperty("javax.rmi.ssl.client.enabledCipherSuites",
-                                   "dummy_ciphersuite");
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory());
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg3);
-                System.exit(1);
-            } catch (Exception e) {
-                System.out.println(msg2 + " Exception: " + e.toString());
-                System.exit(0);
+                        "dummy_ciphersuite");
+                testRmiCommunication(new ServerFactory(), true);
             }
-            break;
-        case 7: /* client side dummy_protocol */
-            System.out.println(msg1);
-            try {
+
+            /* client side dummy_protocol */
+            case 7 -> {
                 System.setProperty("javax.rmi.ssl.client.enabledProtocols",
-                                   "dummy_protocol");
-                HelloImpl server = new HelloImpl(
-                          0,
-                          new ClientFactory(),
-                          new ServerFactory());
-                Remote stub = server.runServer();
-                HelloClient client = new HelloClient();
-                client.runClient(stub);
-                System.out.println(msg3);
-                System.exit(1);
-            } catch (Exception e) {
-                System.out.println(msg2 + " Exception: " + e.toString());
-                System.exit(0);
+                        "dummy_protocol");
+                testRmiCommunication(new ServerFactory(), true);
             }
-            break;
-        default:
-            throw new IllegalArgumentException("invalid test number");
+
+            default ->
+                    throw new RuntimeException("Unknown test number: " + testNumber);
         }
     }
 
-    public static void main(String[] args) {
+    private static void testServerFactory(String[] cipherSuites, String[] protocol, String expectedMessage) {
+        try {
+            new ServerFactory(SSLContext.getDefault(),
+                    cipherSuites, protocol, false);
+            throw new RuntimeException(
+                    "The expected exception for "+ expectedMessage + " was not thrown.");
+        } catch (NoSuchAlgorithmException exc) {
+            throw new RuntimeException("Could not create SSLContext.", exc);
+        } catch (IllegalArgumentException exc) {
+            // expecting the exception for unsupported ciphersuite,
+            // anything else is an error
+            if (!exc.getMessage().toLowerCase().contains(expectedMessage)) {
+                throw exc;
+            }
+        }
+    }
```

Comment:
Overall this test method makes sense, as it asserts that an IllegalArgumentException must be thrown with the given exception message. (A comment to this effect would be helpful.) As such, the catch of NoSuchAlgorithmException is confusing. I think this is here because SSLContext.getDefault() lists this as a checked exception. If so, then I'd suggest simply propagating it by adding a `throws` clause for this method. This also means adding a `throws` clause to the `runTest()` method, possibly just `throws Exception`. In general this is OK for jtreg tests to propagate any checked exception out of main, since jtreg will handle and report any unexpectedly propagated exceptions. It's thus unnecessary for test code to have catch-clauses for anything that the test isn't actually handling.

Reply:
The exception handling has been cleaned up here and the other place you pointed out.